src/engine/controller/Controller.java

```diff
@@ -41,57 +59,56 @@ public ProtoUserDataStream receiveUserRequest(ProtoUserDataStream data) {
     	// don't pass the value 10, pass the value actually want
     	// also need logic to split this into many many requests
     	
+    	final int MAX_THREADS = 50;
+    	
     	if (data == null || data.getInput().size() < 1) {
     	    throw new IllegalArgumentException("Input data can't be null");
     	}
     	
     	Iterator<Integer> dataIt = data.getInput().iterator();
+    	ExecutorService threadPool = Executors.newFixedThreadPool(MAX_THREADS);
+    
+		
+		List<Future<?>> futures = new ArrayList<>();
+    	
     	while(dataIt.hasNext()) {
```

Comment:
I think this should be a for loop of size userInput.

Reply:
Has to be an iterator if the input size is > i32.size